arangod/Graph/GraphOperations.cpp

```diff
@@ -815,10 +899,10 @@ OperationResult GraphOperations::removeEdgeOrVertex(const std::string& collectio
 
       arangodb::aql::Query query(false, _vocbase, queryString, bindVars,
                                  nullptr, arangodb::aql::PART_DEPENDENT);
-      query.setTransactionContext(ctx); // hack to share  the same transaction
+      query.setTransactionContext(ctx);  // hack to share  the same transaction
```

Comment:
Does anyone know why this requires a "hack", or whether this hack is still there in `devel`?

Reply:
@markuspf I do not know in detail. We need to ask @graetzer, he added it,